deployment/Dockerfile

```diff
@@ -0,0 +1,137 @@
+# service.
+#
+# ---------------------
+# ENVIRONMENT VARIABLES
+# ---------------------
+#
+# The docker image assumes that certain environment variables are defined at
+# run-time. See `dts.yaml` for a list of these variables and an illustration
+# of how they are used.
+#
+# ------------------
+# BUILDING THE IMAGE
+# ------------------
+#
+# To build the image, issue the following commands from the top-level
+# directory:
+#
+# 1. docker build --network=host [--build-arg ARG1=VAL1 ...] -t dts .
+# 2. docker image prune --filter label=stage=builder
+#
+# The first command builds the image using an intermediate "builder" image.
+# The second command deletes the builder image.
+
+#-------------------------------------------------------
+#           Dockerfile content starts here
+#-------------------------------------------------------
+
+# Here's a reasonable Go-equipped Docker image.
+FROM golang:1.21-alpine AS builder
+
+# Process build arguments and pass them along to the environment.
+ARG CONTACT_NAME=Unspecified
+ARG CONTACT_EMAIL=dts@example.com
+ARG TERMS_OF_SERVICE_URL=http://example.com
+ARG SERVER_URL=http://example.com
+
+#-------------------------------------------------------
+# Build the service executable.
+#-------------------------------------------------------
+# We label this image as "builder" so we can prune it afterward with
+# docker image prune --filter label=stage=builder
+LABEL stage=builder
+# We need Git, Curl, Zlib, and npm.
+RUN apk update && apk add --no-cache git curl build-base zlib-dev zlib-static npm
+# Set the directory we work in within the image.
+WORKDIR /go/dts
+# Copy files from our context to the working directory.
+COPY . .
+# Use redoc-cli to generate API documentation
+RUN npm install -g redoc-cli
+RUN go mod vendor
+RUN go generate
+# Build dts as a static executable with documentation endpoints.
+RUN env CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -tags docs -a
+
+#--------------------------------------------------------
+# Now we create a smaller image containing only
+# 1. the service executable
+# 2. a generated configuration file (dts.yaml)
+# 3. other UNIX stuff provided by the `alpine` base image
+#--------------------------------------------------------
+
+# Here's the alpine Docker container. If you ever want to peek at its contents,
+# type `docker run -it alpine /bin/ls`.
+FROM alpine:3.19
+
+# Process environment variable build arguments
+ARG USERNAME=gsuser
+ENV USERNAME=$USERNAME
+ARG UID=1000
+ENV UID=$UID
+ARG GROUP=gsgroup
+ENV GROUP=$GROUP
+ARG GID=1001
+ENV GID=$GID
+
+# Copy essential stuff from the builder image.
+COPY --from=builder /go/dts/dts /bin/dts
+COPY --from=builder /go/dts/deployment/dts.yaml /etc/dts.yaml
+
+ENV PATH=/bin:/usr/bin:/usr/sbin
+
+# Add the user and group for this image and set it up.
+RUN /usr/sbin/adduser -u $UID -H -D $USERNAME
+RUN /usr/sbin/addgroup -g $GID $GROUP
+RUN /usr/sbin/addgroup $USERNAME $GROUP
```

Comment:
if you want to be super efficient and reduce the number of layers in your docker image, you can combine all these RUN statements into a single one (i.e. `command_1 && command_2 && command_3`)